### ISSUE TITLE
Alt+Tab: show all window previews

### ISSUE THREAD

**Issue by donadigo** [elementary/gala#72]
While the current presentation of windows in trunk's Alt+Tab got better I think we should consider redesigning the Alt+Tab behaviour further. The main problem with the current behaviour:

**Windows are covered by themselves**: this leaves user without knowledge what windows he has opened and how do they look like. If he wants to know, he has to cycle through all of them. Currently we have a dock-like clone on the bottom that shows the current applications open but this solves issue only partially, there could be multiple windows attached to the application. This also:
* forces the user to drastically shift the focus from the center from the center to the bottom of the screen
* is one of the issues why we can't expose a setting to change dock's position to left or right position of the screen
* you cannot choose a window with a mouse when alt+tabbing, because the windows are covered

It is significantly faster and easier (at least for me) to switch to a desired window when I see it immediately after pressing Alt+Tab. You can easily see how much time it takes for you to  recognize the window you want to switch to: just compare the current Super+A (expose all windows) behaviour with the Alt+Tab one. Please discuss in the comments.

**Comment by Philip-Scott** [comment]
One way that could probably speed up Alt+Tab would be to allow scrolling across the opened apps, just like you can scroll across workspaces in the overview

**Comment by peteruithoven** [comment]
It's a bit hard to talk about this without a visual. 
Shouldn't a user just  use expose if he likes that specific behavior more? 
( "the current presentation of windows in trunk's Alt+Tab" not been released to non daily users [yet](https://github.com/elementary/gala/pull/53#issuecomment-324318791). )

**Comment by donadigo** [comment]
> It's a bit hard to talk about this without a visual.

That's why there's "needs design" label.

> Shouldn't a user just use expose if he likes that specific behavior more?

Expose is not alt+tab. You can't cycle through all the windows quickly with keyboard (although we could take that into consideration).

**Comment by alexbhandari** [comment]
I've been looking into this and I stumbled across this video: https://www.youtube.com/watch?v=TJDepv2ftU4 which seems to address what is discussed here (and provides a visual). Did gala ever function like this where all windows were visible during alt-tab?

For reference the current behavior in gala that I have is like this: https://www.youtube.com/watch?v=l9tcsOjUgK0 which makes it hard to put the focused window in context with the rest of the windows because all other windows disappear during alt-tab.

**Comment by matzipan** [comment]
I like what Ubuntu has done in Gnome: https://didrocks.fr/2017/09/14/ubuntu-gnome-shell-in-artful-day-12/

**Comment by Godje** [comment]
I think for the better scroll/view of the windows there is a `Super+T` shortcut, which literally is named `Show All Windows`. If a user of the OS is introduced to such a handy shortcut, he/she might as well have a choice between a quick `Alt+Tab`, or complete view of `Super+T`

**Comment by Nine-H** [comment]
@alexbhandari there's a vestigial gsettings key at `org.pantheon.desktop.gala.appearance alt-tab-window-opacity` that controls how visible background windows are during alt tab. it doesn't really help with windows occluding each other, and it's a little janky when depth order changes.

**Comment by alexbhandari** [comment]
@Nine-H thanks! I find that with the increased opacity alt-tab is more intuitive, but I see what you mean in terms of it being a little janky in some instances

**Comment by donadigo** [comment]
I've pushed an experimental, prototype branch here if anyone wants to test it out: https://github.com/donadigo/gala-alt-tab-window-preview

**Comment by mahajanudit** [comment]
I used @donadigo and I like the concept better than the original. But I found few issues with it (may be better call them quirks)
1. Thumbnail size difference looks just not right. Off course different apps will have different windows size but to make things look consistent we can have a dimension (preferably height) as same and scale the windows of the thumbnail to make windows look more consistent in the switcher.
2. Too many thumbnails cause the switcher to over flow out of the screen. We should have a limit to the number of windows shown in the screen of the switcher and if that exceeds, it should auto scroll. Off course there should be no scrollbar, the scrolling should be intuitive so as to make sure currently focused windows on the switcher remains somehow at the center of the screen. 

But this is a really good start

**Comment by Blast-City** [comment]
What if the windows would be grouped by application, but unlike other switcher the windows of same apps would be stacked like cards. [Like this](https://www.outsystems.com/blog/-/media/Images/Blog/Posts/gestures_glamour_swipeable_stacked_cards/gestures_glamour_swipeable_stacked_cards_04.png?h=418&w=750&la=en&hash=47F488C63CB2AD400649D26ABF261E00DF1B0DC2)

Alt+tab and left/right would select app and up/down would select window in stack.

**Comment by attila123** [comment]
Hi, sorry, did not read all messages, just tried Pantheon and the Alt + Tab experience is stopping me for switching to that. I have a laptop with 2 external monitors and it's very difficult to see which application would activate. I would like a behavior similar to Cinnamon (at least how it is set up for me now): when pressing the Alt + Tab in Cinnamon, the window list appears in that monitor where the mouse is. This is very user friendly and it is a must for me! Uploading a screenshot how it looks like.
![alt-tab-switcher-cinnamon](https://user-images.githubusercontent.com/1728046/54343543-57b6ea00-463f-11e9-8e2f-98bf03406cea.png)

**Comment by Godje** [comment]
@attila123 it is a must for you, but not necessarily for everyone. People use what they want here in Linux. Whatever works for you, use it. So if you like it the way you have it, why bother switching to Elementary, if you got what you need. Just install Plank, and configure your bar to have similar layout to Elementary.

**Comment by attila123** [comment]
@Godje Maybe you misunderstood the screenshot, it is *not* a dock panel, but the window switcher "dialog" which pops up when I do Alt + Tabbing.
I just ask the Pantheon developers to consider adapting useful ideas, such as displaying the window switcher dialog in the display where the mouse is.
Yes, currently Cinnamon serves my needs well (especially with changing number of displays, docking, undocking, etc, btw as per my quick testing Pantheon seems to do good in this area), however the "Linux world" is also about the availability of alternatives, and so I would like to see working alternatives to Cinnamon.

**Comment by Godje** [comment]
@attila123 i can understand, considering how nice Elementary is in looks and Vala apps integration.
I understood you are talking about an Alt+Tab menu. I just pointed out that "this is a must for me" to switch to elementary, may be what you want from the OS, but not the rest. I don't know how many people want it to be the way you describe it. But whatever they have the moment, I would like them to implement display of Alt+Tab on the window where the mouse is too.
I don't know why this issue is still open, running since September 2017

**Comment by Blast-City** [comment]
> Hi, sorry, did not read all messages, just tried Pantheon and the Alt + Tab experience is stopping me for switching to that. I have a laptop with 2 external monitors and it's very difficult to see which application would activate. I would like a behavior similar to Cinnamon (at least how it is set up for me now): when pressing the Alt + Tab in Cinnamon, the window list appears in that monitor where the mouse is. This is very user friendly and it is a must for me! Uploading a screenshot how it looks like.
> ![alt-tab-switcher-cinnamon](https://user-images.githubusercontent.com/1728046/54343543-57b6ea00-463f-11e9-8e2f-98bf03406cea.png)

In the image you posted, are this icons corresponding to the open windows in the second monitor or all the applications on all monitors/workspaces?

**Comment by attila123** [comment]
@Blast-City The window icons correspond to all the open windows. Displaying the window list where on the monitor where the mouse is is just a convenience feature as the user is probably looking that monitor. But displaying the window list on all monitors is also as good, I think.

**Comment by Blast-City** [comment]
> @Blast-City The window icons correspond to all the open windows. Displaying the window list where on the monitor where the mouse is is just a convenience feature as the user is probably looking that monitor. But displaying the window list on all monitors is also as good, I think.

Thanks. 

I was just trying to understand how it works on Mint.

The problem is that on Elementary the window switcher have two components. The windows on the to and icons in the bottom. So both solution (yours or the current one) will run into the issue where you don't see feedback on the top (windows portion of the switcher).

I think your solution would work great if the second monitor would be treated as an independent workspace. When your mouse would be on the second monitor, the switcher would only show the windows on that monitor. The same way that it works on a single monitor when a workspace is selected.

**Comment by Blast-City** [comment]
Any news on this?

One thing that I find hard on elementary is finding the right window when you have multiple windows on multiple workspaces. I always have to use the workspace switcher plus the window switcher.

Having an Alt+Tab window switcher that would show windows across all workspaces with a window preview would make the above problem disappear and it could potentially make Super+A and Super+W obsolete.

**Comment by janxkoci** [comment]
Well I use workspaces exactly in the oposite way - to get less windows in my alt+tab to prevent endless switching. I usually have certain windows on the same workspaces everytime, but if I can't find some window, I do super+A overview of all windows or look at the dock. Also multitasking view is sometims good to find windows from other workspaces.

The current alt+tab behavior is one of the reasons I love elementary so much.

And I think the original reason to open this issue was already implemented in Juno (as I understand from the description of the problem) so this issue can be closed IMO.

**Comment by danrabbit** [comment]
I think using the dock seemed clever at the time because we were removing elements from the display instead of adding elements, but I would really like to get away from associating the dock with multitasking and it's not necessarily an accurate representation of windows as much as it represents apps.

So I agree with not relying on the dock to show the order of tab switching.

I also agree that leaving windows where they are makes it difficult to see what will be switched to next. However, I think one thing that's nice about the current design is that it's very clear which window is currently on top before you exit the view.

I'm playing with the prototype that Adam proposed and I'm finding myself struggling a bit to see which window is actually highlighted and where my current selection is.

It's also kind of jarring to me that windows aren't displayed in a linear fashion here. I think it would help make it clearer what comes next if windows were laid out in either a column or row (I think row is probably preferable here).

I'm wondering if something more like iOS or UbuntuPhone's multitasking view might make sense here so we can get window previews as close to 1:1 as possible.

Maybe something in this direction, but with more spread so we could see more of the windows underneath:

![https://cdn.dribbble.com/users/128446/screenshots/4890550/appswitch.gif](https://cdn.dribbble.com/users/128446/screenshots/4890550/appswitch.gif)

---

btw I was googling and found this which seems kind of interesting: http://www.azarask.in/blog/post/solving-the-alt-tab-problem/

**Comment by Nine-H** [comment]
the cinnamon desktop has an implementation of card stack window switching you can use by binding a shortcut to cycle windows, as did windows vista which had it set to <super>+<tab>.

**Comment by janxkoci** [comment]
Really? This is exactly what Windows 7 does after pressing Super+Tab.

**Comment by donadigo** [comment]
@danrabbit I'm willing to try implementing stacking cards, its an interesting idea, although I'm not sure if it would work in this setting. The link is actually something that I've been trying to solve for workspaces for a while now. It seems like it could be applied here too, but I will first look at the redesign, since I think this is a bigger problem.

**Comment by Blast-City** [comment]
I've been using the Super+S shortcut to switch windows/workspaces and thought of this:

What if Alt+Tab and Super+Tab would both show Workspace overview if pressed for some time.

Alt+Tab would switch windows on the workspace overview.
Super+Tab would cycle through the workspaces on the workspace overview.
Super+S/Super+Down would show workspace overview and Alt+Tab/Super+Tab could be used to switch windows/workspaces without exiting workspace overview.

So every time workspace overview was open the currently window would be already selected. Since it shows an overview of windows and workspaces, an user could easily tell where all the things are placed.

**Comment by janxkoci** [comment]
@Blast-City that would really mess up my workflow - I use Super+Tab for quick switching between workspaces, when overview is too slow. Merging everything into one is not helpful in any way imho.

**Comment by Blast-City** [comment]
> @Blast-City that would really mess up my workflow - I use Super+Tab for quick switching between workspaces, when overview is too slow. Merging everything into one is not helpful in any way imho.
> 
> My current workflow usually works like this:
> 
>     * If I have too many windows that get in the way during Alt+Tabbing, I either
>       
>       * move some of them to another workspace
>       * minimize some of them to put them at the end of Alt+Tab list
>       * close them (e.g. email, from where I opened some file)
> 
>     * If I have several windows arranged in certain view (e.g. side-by-side) and need to work with another window too, I put it on the second workspace and switch between them with Super+Tab (it's like Alt+Tab between groups of windows, super cool)
> 
>     * If I need some part of window to be visible with other windows, I put it on other workspace or minimize it and use PIP to see the part I need. PIP allows dragging with mouse from any part, so it's way more convenient than e.g. resizing the window and moving it with titlebar or something like that. Not to mention my OCD hates resizing windows that already have nice size for most uses rofl
> 
> 
> In my view elementary OS already has really good set of tools to handle situations with many windows.

Hi.

You would still be able to use Super+Tab the way you use since a fast Super+Tab doesn't bring the overview.
The problem with the Super+Tab is that to get to the workspace you want it goes through all your workspaces and if you miss it you have to go through all of them again. Showing a visual indication (workspace overview) would solve this. You would exactly know what comes next, where you are, etc...

As for the Alt+Tab I agree with you, but I think that showing the workspace overview would solve the issue in this thread while maintaining a familiar elementary look. When people look at a workspace overview picture they immediately know what OS it is. And again, fast Alt+Tab would not bring the overview up.

**Comment by janxkoci** [comment]
> You would still be able to use Super+Tab the way you use since a fast Super+Tab doesn't bring the overview.

Not really - switching between more workspaces would require holding Super and pressing Tab, which would probably trigger the overview (I assume). This is a slowdown for me, unless I misunderstand how you mean it.

> The problem is that, right now, to get to the workspace you want you have to go through all your workspaces and if you miss you have to go through all of them again. Showing a visual indication (workspace overview) would solve this. You would exactly know what comes next, where you are, etc...

I don't really have this problem. First, I know my workspaces most of the times. And switching with Super+Tab is very fast, I have no problem cycling through 4 workspaces to find the one I want. **If I don't know where to go, I simply don't use Super+Tab but something else** (either overview or show all windows with Super+A).

Moreover, in some workflows I actively limit the number of my workspaces to 2, so I can quickly switch between them with Super+Tab - especially when I have multiple windows in certain arangement on each of the two workspaces. This is one of the best features of the OS and I was missing it everywhere else (GNOME, macOS, ...). And solving this by making the shortcut behave differently if used in a short or long way will be so unintuitive that most people would probably never discover that such feature is available.

**In short:** There are different tools for different situations, I really don't see the point in making one of the tools do everything.

> As for the Alt+Tab I agree with you, but I think that showing the workspace overview would solve the issue in this thread while maintaining a familiar elementary look. When people look at a workspace overview picture they immediately know what OS it is. And again, fast Alt+Tab would not bring the overview up.

I do like the look of elementary OS a lot. But I also like how it works and the proposed changes would seriously alter the way I use it now. Possibly up to the point to make it unusable for me, forcing me to search different OS. And as I recently abandoned all other OSes and switched to elementary full-time, I would like to not have to do that.

**Comment by Blast-City** [comment]
> > You would still be able to use Super+Tab the way you use since a fast Super+Tab doesn't bring the overview.
> 
> Not really - switching between more workspaces would require holding Super and pressing Tab, which would probably trigger the overview (I assume). This is a slowdown for me (not to mention getting dizzy from all the zooming of all windows), unless I misunderstand how you mean it.
> 
> > The problem is that, right now, to get to the workspace you want you have to go through all your workspaces and if you miss you have to go through all of them again. Showing a visual indication (workspace overview) would solve this. You would exactly know what comes next, where you are, etc...
> 
> I don't really have this problem. First, I know my workspaces most of the times. And switching with Super+Tab is very fast, I have no problem cycling through 4 workspaces to find the one I want. **If I don't know where to go, I simply don't use Super+Tab but something else** (either overview or show all windows with Super+A).
> 
> Moreover, in some workflows I actively limit the number of my workspaces to 2, so I can quickly switch between them with Super+Tab - especially when I have multiple windows in certain arangement on each of the two workspaces. This is one of the best features of the OS and I was missing it everywhere else (GNOME, macOS, ...). And solving this by making the shortcut behave differently if used in a short or long way will be so unintuitive that most people would probably never discover that such feature is available.
> 
> **In short:** There are different tools for different situations, I really don't see the point in making one of the tools do everything.
> 
> > As for the Alt+Tab I agree with you, but I think that showing the workspace overview would solve the issue in this thread while maintaining a familiar elementary look. When people look at a workspace overview picture they immediately know what OS it is. And again, fast Alt+Tab would not bring the overview up.
> 
> I do like the look of elementary OS a lot. But I also like how the OS works and _some_ of the proposed changes would seriously alter the way I use it now. Possibly up to the point to make it unusable for me, forcing me to search different OS. And as I recently abandoned all other OSes and switched to elementary full-time, I would like to not have to do that.

The idea would be to press fast when you wanted to go to the last window selected. If an user needed to go to another window (press more than two times), it would give a visual cue (overview). The zooming argument is interesting. I didn't thought of that. You're probably right.

One thing I'm sure is that it would still be cool to have the ability to use Alt+Tab and Super+Tab on the multitasking view.

**Comment by janxkoci** [comment]
> The idea would be to press fast when you wanted to go to the last window selected. If an user needed to go to another window (press more than two times), it would give a visual cue (overview). 

But that's what I meant - sometimes I skip multiple workspaces with Super+Tab, because it's very fast and I can do it with one hand. If it opened overview, that would be slower and quite distracting, imho.

**Comment by scottjenson** [comment]
Just for context the two most common alt-tab switchers are Mac and Window. I'm *not* advocating we copy them, only that they each provide a example we can use to discuss and critique.

Windows:
- Shows all windows of all apps
- Layout is in a horizontal row
- Clear highlighting as you move through each one
- Can also go backwards, use mouse, quit apps (with mouse click)

Mac
- Shows only App icons (sub windows not shown) (This is much faster)
- Layout is also in a horizontal row
- Clear highlighting as you move through each app
- Can also go backwards, use the mouse, and even quit apps (with keyboard)

Elementary Today
- No abstract layout of apps or windows. it's more of a "highlighting" of the next app in the chain.
- you move from windows to window in one direction

The differences when laid out like this are pretty clear: elementary is more of a "1-at-a-time switcher" while Win/Mac offer more of an "Overview Picker".  Here is what I heard above that people would like to see:
- Show all windows (not just apps) in the current workspace
- Not icons but scaled down images of each window (but will this slow things down?)
- Arrange in a more horizontal layout (there could be overlapping, needs to be explored)
- All windows w/in a single app could be grouped/tiled together
- Clearly highlight (or even zoom slightly) as you move from window to window
- Still maintain the 'quick switch' behavior of Alt-Tab so you can toggle between two apps repeatedly. 

I'm happy to mock up a range of visuals (not a single solution) to be discussed further. But it's clear there are a lot of constraints. Anything I missed?

**Comment by janxkoci** [comment]
Nice overview :)

> All windows w/in a single app could be grouped/tiled together

I just want to note that, in my view, grouping windows of same app under one icon in switcher will completely break my alt-tabbing workflows and I would have to abandon elementary and look for something else (I don't believe the devs would implement a toggle for this, as the two models are quite different in how they behave). I used to disable grouping wherever it was possible (e.g. GNOME) and really loved the fact it's not present in elementary.

**Comment by scottjenson** [comment]
Here are my first two attempts. I'm intentionally keeping the animations simple to keep our discussion focused. I've called out additional directions to explore at the end.

Here's the first one: Basic app to app transition:
![Simple Transition](https://user-images.githubusercontent.com/1055674/90988045-82031b80-e544-11ea-95e8-245187f68ad2.gif)
A few notes:
- Each app is it's own color just to make it easier for you to track
- The title is below each window icon
- This animation is a simple "alt-tab" toggle switch
- If you wanted to go to the final app, you'd just hit alt-tab again

Here's a small addition, showing multiple windows for an app:
![Simple Transition 2](https://user-images.githubusercontent.com/1055674/90988183-deb30600-e545-11ea-8ef3-3644f650a232.gif)

What I'm trying to build here is an agreement on how we'd like this new switcher to work. The key things we're getting with this model are:
- A clean, fast animation to get an overview of all windows
- A backward compatible Alt-tab approach to quickly toggling between 2 windows
- A way to extend that to multiple windows/apps
- A future path to allow additional functionality (quitting apps from this view)

For me the critical questions are:
- Are there any performance issues with this approach?
- Does this *roughly* capture what the community wants?

Once we get agreement on that, we can tackle the important edge cases:
- What happens with odd shaped windows (e.g. mostly vertical or horizontal)
- What happens with lots of windows (go smaller? Wrap to second row?)

**Comment by scottjenson** [comment]
I appreciate this can't *just* be visually pleasing, it also has to be efficient and quick. It's not an accident that both Mac and Win have zero animation: it's much faster. I'm hoping we can have style **and** speed.

Some obvious things to keep the animation but make it faster:
- reduce the animation speed. It's currently 500ms but we could likely make it much faster e.g 250ms.
- Don't animate the window name from CURRENT to TARGET. Just start on TARGET. Only animate the window name if you hit Alt-Tab a second time.
- Short circuit the animation. If you tap and release Alt-Tab quickly, just switch apps with no animation. This more of a power user feature. The Mac does this, Win does not

**Comment by cassidyjames** [comment]
@scottjenson thanks for tackling this! This looks like a great starting point, and I _really_ appreciate the motion mockups; that makes it so much easier to track and discuss.

My main concern so far is what you've addressed regarding the speed; just tapping Alt+Tab to get to the last-focused app seems to be really common for workflow anecdotally and personally. I think reducing the animation is probably necessary in general, but dropping/significantly speeding it up on release of both Alt and Tab is probably a good in-between while keeping the Alt+Tab+Tab+Tab to switch through multiple windows feeling nice.

As a sort of side-note, we use Material Design inspired values for widget [animations in the stylesheet](https://github.com/elementary/stylesheet/blob/be9ecefc174e9286ffeeab4e1b850cc7e758a1e7/src/_animate.scss#L1-L17). I am not sure what Material generally says about these sort of larger animations, but I do feel like generally we keep things around or under 300ms to feel snappy. Here are the current constants used in Gala: https://github.com/elementary/gala/blob/49d1d8ffabeb5fcd79f9d424adc21e8ad8f1fec6/lib/Constants.vala#L20-L33

My other main concern is around the size of the previews; if I have two large Code windows open, they're gonna look almost identical at that size. I wonder if we should consider larger previews with some overlap like in @danrabbit's previous mockup. I'm pretty open to discussion, though, as that would consequently mean fitting fewer windows on screen at once _and_ obscuring parts of the window previews in the trade-off for larger previews.

> Are there any performance issues with this approach?

I think this would be as performant as our other window animations, e.g. the workspace-focused multitasking overview. I don't see any inherent issues with it.

> Does this roughly capture what the community wants?

Hard for me to answer personally, but I _suspect_ so. A common complaint I hear is that the order feels unpredictable—I think because people don't see the dock or associate the disconnected app icons strongly with the windows themselves, and our current window previews don't indicate the order at all. This pretty neatly solves it by making the window previews linear. It does also feel more similar to other platforms and like it's not trying to be "clever" by re-using the dock.

>A future path to allow additional functionality (quitting apps from this view)

I think that's actually probably the wrong direction specifically… we have the multitasking overview which gives you a complete spread of open windows and allows for quick closing. I think trying to hold Alt+Tab and then perform actions is just going to get overcomplicated. But that's future talk, anyway. :)

**Comment by scottjenson** [comment]
After discussing this in the Slack channel we discussed a few next steps:

- Shorten the animation time
- Use the app icon + name at the top
- Have the window be below this on a grid
- Have different sized windows shrink to fit this grid (see how this looks for various sizes)
- Much discussion around many windows: scroll to the right or have multiple rows?fffffffffff (tdb)
- Additional discussion of using Alt-Tab vs Multi-Tasking view, & when to use each one (needs more work)

Action item for me was to use real windows this time and create a 'next step' animation and see how it feels. Rinse, repeat.

**Comment by scottjenson** [comment]
OK, here is my next version, following a layout suggestion from @danrabbit 
![Elementary OS Switcher 2 (left)](https://user-images.githubusercontent.com/1055674/91641471-78146900-e9d9-11ea-88a0-b779fcb732cf.gif)

Changes from previous version:
* Real window captures
* Real window sizes (but note, not extreme windows sizes yet, that's coming next)
* Each windows is in the same size grid (which takes up more space (but feels more pleasing, uniform)
* MUCH faster animation
* I tried windows that ONLY have an app title and those that have varying titles
* Note: the dock animates out of the way while switching, not sure what it's value is?

This feels much faster yet keeps a nice "ElementaryOS" look and style to it (but I'm new here so please give feedback on the visual design) What I will say is that it's fairly LARGE and that feels like the next issue we should discuss. Should the Alt-Tab window be only the last X windows and we optimize for that? Or should this scale to 20 windows? That seems like the next core question for us to answer.

**Comment by cassidyjames** [comment]
@scottjenson I'm really liking where this is heading. I think I'm confident enough in this direction that we could start prototyping it—@davidmhewitt wants to rework window switching for technical reasons, anyway, so starting here is better than wasting time re-making the current design.

I think we'll want to pay attention to and tweak on thumbnail sizes, animations speeds and curves, etc. But that can be done relatively easily with a working prototype.

Something I do think we should consider is whether all apps should be scaled evenly (i.e. all at 10% of their original size or whatever they show here), or if they should scale to fit within a certain area. I am leaning toward the latter; in your example, see Calculator. To me, it feels awkward that it could be both taller and wider to better fill that space, which would probably help identifying some smaller windows especially. But perhaps that will also come when addressing the extremes.

This may just be an artifact of the prototype, but window labels should all be left-aligned in this layout.

Just as an aside, the OS does have a user-selectable "accent color" we can use in these designs; it's the blue palette by default, but a user can choose between any of the colors here:

![blue, mint, green, yellow, orange, red, pink, purple, brown, slate](https://user-images.githubusercontent.com/611168/97354409-ee5b0d80-185a-11eb-809c-e4229c458aed.png)

If you find yourself needing _a color_ (especially for representing highlighted or active things), this might be a nice place to use a subtle accent to bring that color more into their experience.

**Comment by danrabbit** [comment]
here's a quick mock showing apps trying to take up the same width instead of scaling at the same rate:

![mock](https://user-images.githubusercontent.com/7277719/97372042-e35ea800-186f-11eb-882c-e948881110d4.png)

**Comment by kmandarin-orange** [comment]
I do feel these mock-ups don't feel very at-home in elementary. Combining some existing design elements, like Show All Windows layout and current window switcher's shading on unfocused apps might help ground this in the system better. I've got a quick sketch here, I'm not much of a Gimp pro haha.

![20201109_130632](https://user-images.githubusercontent.com/55755379/98593648-12761000-2299-11eb-8b86-4be4d2be0d31.jpg)

The main details I think make this fit in well are

- All apps are scaled like @cassidyjames suggested
- Zoom in on focused application, slightly overlapping any applications next to it 
- Icons will not zoom but stay on bottom middle of apps
- Selected app has window name in label beneath in same style that shows up in the rest of the system, I think it's called overlay bar?
- Subtle shade on other application windows/icons
- Slightly darker shade on background
- Grid layout similar to Gnome overview/Show All Windows view in eOS
- Like current switcher, clicking on icons/windows in this view will instantly switch to them
- I really like the movement/animation in @scottjenson's mock, though faster may be better
- Dock dodges out of the way like the workspace overview animation does

Something I didn't illustrate right was Wingpanel transparency, which should just take the system default. @danrabbit's mock does this well. Anyway, that's what I have to offer here.
